src/server/http/index.js

```diff
@@ -10,13 +11,25 @@ module.exports = function (kbnServer, server, config) {
   server = kbnServer.server = new Hapi.Server();
 
   // Create a new connection
-  server.connection({
+  var connectionOptions = {
     host: config.get('server.host'),
     port: config.get('server.port'),
     routes: {
       cors: config.get('server.cors')
     }
-  });
+  };
+
+  // enable tls if ssl key and cert are defined
+  if (config.get('server.ssl.key') && config.get('server.ssl.cert')) {
+    _.merge(connectionOptions, {
```

Comment:
Nope, no reason. I actually had it that way to begin with, really not sure why I felt compelled to change it...